oss-internship-2020/libtiff/example/main_sandboxed.cc

```diff
@@ -0,0 +1,305 @@
+                            int min_alpha, int max_alpha,
+                            const sapi::v::Array<unsigned>& buffer) {
+  // RGBA images are upside down - adjust for normal ordering
+  int adjusted_pixel = pixel % 128 + (127 - (pixel / 128)) * 128;
+  uint32 rgba = buffer[adjusted_pixel];
+
+  if (TIFFGetR(rgba) >= (uint32)min_red && TIFFGetR(rgba) <= (uint32)max_red &&
+      TIFFGetG(rgba) >= (uint32)min_green &&
+      TIFFGetG(rgba) <= (uint32)max_green &&
+      TIFFGetB(rgba) >= (uint32)min_blue &&
+      TIFFGetB(rgba) <= (uint32)max_blue &&
+      TIFFGetA(rgba) >= (uint32)min_alpha &&
+      TIFFGetA(rgba) <= (uint32)max_alpha) {
+    return absl::OkStatus();
+  }
+
+  return absl::InternalError(absl::StrCat(
+      "Pixel ", pixel, " did not match expected results.\n",
+      "Got R=", TIFFGetR(rgba), " (expected ", min_red, "..", max_red,
+      "), G=", TIFFGetG(rgba), " (expected ", min_green, "..", max_green,
+      "), B=", TIFFGetB(rgba), " (expected ", min_blue, "..", max_blue, "), A=",
+      TIFFGetA(rgba), " (expected ", min_alpha, "..", max_alpha, ")\n"));
+}
+
+}  // namespace
+
+std::string GetFilePath(const std::string& dir, const std::string& filename) {
+  return sandbox2::file::JoinPath(dir, "test", "images", filename);
+}
+
+std::string GetFilePath(const std::string filename) {
+  std::string cwd = sandbox2::file_util::fileops::GetCWD();
+  auto find = cwd.rfind("build");
+
+  std::string project_path;
+  if (find == std::string::npos) {
+    LOG(ERROR) << "Something went wrong: CWD don't contain build dir. "
+               << "Please run tests from build dir or send project dir as a "
+               << "parameter: ./sandboxed /absolute/path/to/project/dir \n";
+    project_path = cwd;
+  } else {
+    project_path = cwd.substr(0, find);
+  }
+
+  return sandbox2::file::JoinPath(project_path, "test", "images", filename);
+}
+
+absl::Status LibTIFFMain(const std::string& srcfile) {
+  // without addDir to sandbox. to add dir use
+  // sandbox(absolute_path_to_dir, srcfile) or
+  // sandbox(absolute_path_to_dir). file and dir should be exists.
+  // srcfile must also be absolute_path_to_file
+  TiffSapiSandbox sandbox("", srcfile);
+
+  // initialize sapi vars after constructing TiffSapiSandbox
+  sapi::v::UShort h, v;
+  sapi::StatusOr<TIFF*> status_or_tif;
+  sapi::StatusOr<int> status_or_int;
+  sapi::StatusOr<tmsize_t> status_or_long;
+  absl::Status status;
+
+  status = sandbox.Init();
+
+  SAPI_RETURN_IF_ERROR(sandbox.Init());
+
+  TiffApi api(&sandbox);
+  sapi::v::ConstCStr srcfile_var(srcfile.c_str());
+  sapi::v::ConstCStr r_var("r");
+
+  SAPI_ASSIGN_OR_RETURN(
+      status_or_tif, api.TIFFOpen(srcfile_var.PtrBefore(), r_var.PtrBefore()));
+
+  sapi::v::RemotePtr tif(status_or_tif.value());
+  if (!tif.GetValue()) {
+    // tif is NULL
+    return absl::InternalError(absl::StrCat("Could not open ", srcfile));
+  }
+
+  SAPI_ASSIGN_OR_RETURN(status_or_int,
```

Comment:
`SAPI_ASSIGN_OR_RETURN` will assign the variable if StatusOr<X> is ok. Can't you ust use `int status` here instead of `StatusOr<int>`?